src/main/java/hudson/plugins/ec2/util/AmazonEC2FactoryImpl.java

```diff
@@ -14,8 +14,11 @@
 
     @Override
     public AmazonEC2 connect(AWSCredentialsProvider credentialsProvider, URL ec2Endpoint) {
-        AmazonEC2 client = new AmazonEC2Client(credentialsProvider, EC2Cloud.createClientConfiguration(ec2Endpoint.getHost()));
-        client.setEndpoint(ec2Endpoint.toString());
-        return client;
+        if (ec2Endpoint != null && ec2Endpoint.toString().trim().length() > 0) {
```

Comment:
Method should be annotated with CheckForNull

Reply:
Added annotation